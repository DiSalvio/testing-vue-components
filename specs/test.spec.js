import TestComponent from '@/test.vue'
import ListComponent from '@/list.vue'
import { mount } from '@vue/test-utils'

describe('mounted test component', () => {

  test('has correct default prop value', () => {
    const wrapper = mount(TestComponent)

    expect(wrapper.props().value).toContain('Test')
  })

  test('prop value can be set correctly', () => {
    const wrapper = mount(TestComponent, {
      propsData: {
        value: 'You'
      }
    })

    expect(wrapper.props().value).toContain('You')
  })

  test('html matches snapshot with default prop value', () => {
    const wrapper = mount(TestComponent)

    expect(wrapper).toMatchSnapshot()
  })

  test('html matches snapshot with given prop value', () => {
    const wrapper = mount(TestComponent, {
      propsData: {
        value: 'You'
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})

describe('list component', () => {
  test('html lists existing movie data plus any added data', () => {
    const wrapper = mount(ListComponent)
    const movies = wrapper.vm.marvelMovies
    wrapper.setData({ marvelMovies: [ ...movies, 'Endgame' ] })
    expect(wrapper).toMatchSnapshot()
  })
})
