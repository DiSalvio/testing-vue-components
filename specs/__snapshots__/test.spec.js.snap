// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list component html lists existing movie data plus any added data 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
  </ul>
</div>
`;

exports[`list component lists existing movie data plus any added data 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
  </ul>
</div>
`;

exports[`list component mount 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
  </ul>
</div>
`;

exports[`mounted test component html matches snapshot with default prop value 1`] = `
<div>
  <h1>Hello Test</h1>
</div>
`;

exports[`mounted test component html matches snapshot with default prop value 2`] = `
<div>
  <h1>Hello You</h1>
</div>
`;
